### PR TITLE
Add ocean packages to pangeo-notebook

### PR DESCRIPTION
**Pull request by TomAugspurger**
xref https://github.com/pangeo-data/pangeo-docker-images/issues/28

### CODE REVIEW

**Comment by TomAugspurger** [comment]
cc @scottyhq & @jhamman.

File: pangeo-notebook/environment.yml [hunk 1]
```diff
@@ -48,6 +52,14 @@ dependencies:
...
  - xesmf
+ - xgcm
+ - xhistogram
+ - xlayers
+ - xmitgcm
  - xpublish
+ - xrft
  - zarr
+ - pip:
+   - fastjmd95
```
Comment (rabernat) [inline pangeo-notebook/environment.yml:65]: As soon as https://github.com/conda-forge/staged-recipes/pull/11980 is merged, we can install this from conda.

**Comment by scottyhq** [comment]
/condalock 

☝️ remember you need this comment to update lock files prior to building. fine by me to add these packages.

**Comment by scottyhq** [comment]
locking environment failed (https://github.com/pangeo-data/pangeo-docker-images/runs/797223398?check_suite_focus=true)  because pip isn't supported by conda-lock (https://github.com/mariusvniekerk/conda-lock/issues/4). 

We got around this in the ml-notebook image by also including a sidecar requirements.txt. https://github.com/pangeo-data/pangeo-docker-images/blob/master/ml-notebook/requirements.txt

The base-image Dockerfile runs a `pip install` command after `conda create`:
https://github.com/pangeo-data/pangeo-docker-images/blob/d4b3c1ed59becdb3ff88f2fc64f3c113a0aba29f/base-image/Dockerfile#L117

**Comment by TomAugspurger** [comment]
@scottyhq fixed the merge conflicts (I was behind when I made my branch).

**Pull request merged**